File: src/get_single_weather.py
import json
import os

import boto3

dynamodb = boto3.resource('dynamodb')
table_name = os.environ.get("TABLE_NAME")


def lambda_handler(event, context):
    weather_id = json.loads(event['pathParameters'])['id']
    table = dynamodb.Table(table_name)
    try:
        response = table.get_item(Key={'id': weather_id})
        return {
            'statusCode': 200,
            'body': json.dumps(response['Item'])
        }
    except table_name:
        return {
            'statusCode': 500,
            'message': "Unable to get item"
        }

File: packages/functions/src/getSingleWeather.ts
import { DynamoDB } from "aws-sdk";
import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";

const dynamoDb = new DynamoDB.DocumentClient();
const tableName = process.env.TABLE_NAME as string
export async function main(
  event: APIGatewayProxyEvent
): Promise<APIGatewayProxyResult> {
  const weatherId = event.pathParameters && event.pathParameters.id
      ? event.pathParameters.id
      : null;
  const weatherItem = await dynamoDb.get({
    TableName: tableName,
    Key: {
      id: weatherId,
    },
  }).promise();
  if (!weatherItem || !weatherItem.Item) {
    return{
      statusCode: 404,
      body: JSON.stringify({
        message: "No weather data found",
      }),
    }
  }
  return {
    statusCode: 200,
    body: JSON.stringify(weatherItem),
  };
}